common/math/geometry/src/solver/polynomial_solver.cpp

```diff
@@ -73,6 +76,10 @@ std::vector<double> PolynomialSolver::solveQuadraticEquation(
   for (const auto candidate : candidates) {
     if (min_value <= candidate && candidate <= max_value) {
       ret.emplace_back(candidate);
+    } else if (std::abs(candidate - max_value) < tolerance) {
+      ret.emplace_back(max_value);
+    } else if (std::abs(candidate - min_value) < tolerance) {
+      ret.emplace_back(min_value);
```

Comment:
Why are you using emplace_back?
There is no reason to use emplace_back on a container whose element type is a scalar type.

Reply:
Change to push back in this commit.

https://github.com/tier4/scenario_simulator_v2/pull/996/commits/27921ff1a68898236e5a34e2f7a8a1b5b3cae242

Reply:
use push_back for std::vector<double> type in PolynomialSolver class and HermiteCurve class.